### PR TITLE
improved error message

### PR DESCRIPTION
**Pull request by mchangxe**
**What this PR does / why we need it**:
Improves kubectl cp error message by checking if the tar command is present in the container first before actually executing and triggering the copy mechanics. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/cp.go [hunk 1]
```diff
@@ -160,6 +162,36 @@ func checkDestinationIsDir(dest fileSpec, f cmdutil.Factory, cmd *cobra.Command)
 	return execute(f, cmd, options)
 }
 
+
+// testTar receives information about the destination container 
+// and determines if the command tar is present in the containe.
```
Comment (nikhiljindal) [inline pkg/kubectl/cmd/cp.go]: containe -> container

File: pkg/kubectl/cmd/cp.go [hunk 2]
```diff
@@ -184,6 +216,20 @@ func copyToPod(f cmdutil.Factory, cmd *cobra.Command, stdout, stderr io.Writer,
 	}()
 
 	// TODO: Improve error messages by first testing if 'tar' is present in the container?
```
Comment (nikhiljindal) [inline pkg/kubectl/cmd/cp.go]: TODO can be removed now :)

File: pkg/kubectl/cmd/cp.go [hunk 3]
```diff
@@ -184,6 +216,20 @@ func copyToPod(f cmdutil.Factory, cmd *cobra.Command, stdout, stderr io.Writer,
...
+	if err := testTar(f, cmd, dest); err != nil {
+
+		exitCode126 := "command terminated with exit code 126"
+		exitCode127 := "command terminated with exit code 127"
+
+		if err.Error() == exitCode126 {
```
Comment (nikhiljindal) [inline pkg/kubectl/cmd/cp.go]: move this to testTar to remove duplication at both places?

**Review by nikhiljindal — COMMENTED** [review]
Add a test to verify that it returns the expected error?

File: pkg/kubectl/cmd/cp.go (hunk 3 above)
Comment (mchangxe) [reply pkg/kubectl/cmd/cp.go]: will do! Somehow forgot about that!

**Comment by mchangxe** [comment]
@nikhiljindal, I dont really understand what is causing gofmt test to fail during pull-kubernetes-verify. In the logs, all code lines, including comments, with the keyword "error" is being highlighted and somehow gofmt just fails without any error messages.

**Pull request closed without merging**